superset/assets/src/visualizations/FilterBox/FilterBox.jsx

```diff
@@ -93,11 +93,28 @@ class FilterBox extends React.Component {
       actions: { setControlValue: this.changeFilter },
     });
     const mapFunc = control.mapStateToProps;
+    const props = {
+      ...this.props,
+      // the control expects datasource attributes to be snake case
+      datasource: this.snakeCaseAttributes(this.props.datasource),
```

Comment:
According to `FilterBox/transformProps.js`, the `datasource` here should already be the `snake_case` one.

Reply:
Sorry, how do you infer that?

`console.log`ing datasource here shows that it has `timeGrainSqla` instead of `time_grain_sqla` (same for other attributes).

Reply:
Hmm. ic. 
I infer from the `transformProps.js` that it passes `rawDatasource` to the chart component instead of `datasource`. `rawDatasource` is supposed to be the `snake_case` one.

Reply:
Ah, thanks for explaining. 

@xtinec did a revert on the casing, let me test this again with her fix.